Removes security access from guards

CatoChristopherMrow opened the pull request:
## About The Pull Request
Adds a new access type for outposts, and gives it to guards. Removes security and security front door access from guards, meaning they can no longer modify security records, set people to wanted, or wander about security (unless they break in)
## Why It's Good For The Game
Guards having the ability to set people to wanted, add records, remove them, and enter a large majority of sec was an unfortunate side effect of how skyrat forced goof to add in departmental guards. Guards are not security, and shouldn't have access to view security records, let alone change them.

This shouldn't affect their ability to order guns, as those are tied to weapon permits.

I've also gone ahead and changed the access for the departmental baton crates, to ensure they can still open them.
## Proof Of Testing
Works on my machine
<details>
<summary>Screenshots/Videos</summary>

</details>

## Changelog
:cl: UvvU
del: removed departmental guards sec access
fix: fixed some guard offices having no proper access set, making them public access.
spellcheck: made a few typos
/:cl:

CatoChristopherMrow commented:
Ignore the horrid mess of commits, please, I'm not very good at github, but I got there eventually.

CatoChristopherMrow commented:
I'm HOPING files changed is all that actually matters.

Sebastionate commented:
There are some items in the Cargo Department that should be purchasable by Dept guard by default, much like the standard baton. These require Sec-Access currently. Restricting these should be noted. 

Furthermore regardless if it was an accident or not, Dept guards have not shown issue in altering Wanted/Sus etc Records, and doing so falsely/wrongly would be treated the same as Sec Officers doing such, i.e if it's to an extent that it violates the rules of playing a believable character who cares about their job, it'd be treated as such. 

tl;dr, as a Dept guard player I do not support removing these accesses in full from the roles, only the physical department of Security. Furthermore it appears you've removed Guards abilities to access Sec, yet not Sec's ability to access Departmental Guard Outposts, this double standard does not make much sense for separating the accesses.

CatoChristopherMrow commented:
As I said in the PR, I already changed the access for purchasing departmental baton purchases. 
 There's also never a time it's NOT falsely, considering guards arn't security and don't really have the authority to do such.

> There are some items in the Cargo Department that should be purchasable by Dept guard by default, much like the standard baton. These require Sec-Access currently. Restricting these should be noted.

As I said in the PR, I already changed the access for purchasing departmental baton purchases. 

> Furthermore regardless if it was an accident or not, Dept guards have not shown issue in altering Wanted/Sus etc Records, and doing so falsely/wrongly would be treated the same as Sec Officers doing such, i.e if it's to an extent that it violates the rules of playing a believable character who cares about their job, it'd be treated as such.

Yes they have, you don't notice is because you don't ever have to deal with it.

> tl;dr, as a Dept guard player I do not support removing these accesses in full from the roles, only the physical department of Security. Furthermore it appears you've removed Guards abilities to access Sec, yet not Sec's ability to access Departmental Guard Outposts, this double standard does not make much sense for separating the accesses.

As for removing the physical access to sec, this is by far the easiest way to do so. Any more and we would have to maintain an entire map worth of doors rather then just a few offices.

Iamgoofball commented:
This was my original vision for Departmental Guards, as per my original implementation at https://github.com/Skyrat-SS13/Skyrat-tg/pull/8524.

👍

ReturnToZender commented:
Holy shit he actually did the work, fuck yeah dude

CatoChristopherMrow commented:
> Furthermore it appears you've removed Guards abilities to access Sec, yet not Sec's ability to access Departmental Guard Outposts, this double standard does not make much sense for separating the accesses.

What are you talking about? I never said this in the PR and fixing this access means that officers can't enter the offices they could earlier. Guard offices are no longer accessible to sec unless they also have the respective department access.

TealSeer commented on code/__DEFINES/access.dm:
```diff
@@ -49,6 +49,8 @@
 #define ACCESS_HOS "hos"
 /// Access for the detective to get into their office, the medical data console, and some other detective-related stuff.
 #define ACCESS_DETECTIVE "detective"
+/// Access for departmental security outposts.
+#define ACCESS_SECURITY_OUTPOST "security_outpost"
```
gonna need to modular comment this

TealSeer commented on code/__DEFINES/access.dm:
```diff
@@ -449,6 +451,7 @@
 	ACCESS_MECH_SECURITY, \
 	ACCESS_SECURITY, \
 	ACCESS_WEAPONS, \
+	ACCESS_SECURITY_OUTPOST, \
```
ditto

TealSeer commented on code/datums/id_trim/jobs.dm:
```diff
@@ -546,6 +546,7 @@
 	)
 	extra_access = list(
 		ACCESS_BRIG,
+		ACCESS_SECURITY_OUTPOST,
```
same with every change in this file

CatoChristopherMrow commented:
I seem to have caused a runtime with adding access to beepskies. I would really appreciate any help with how I caused this and how to fix it.

CatoChristopherMrow commented:
Ok, I THINK I added what you wanted with modular commenting. If you want something else, tell me.

CatoChristopherMrow commented:
Alright, looks like the two failures are from flaky stuff, so unless specific changes are requested this should be good.

Rykka-Stormheart commented:
If they're getting security access revoked, can they be put back in the antag list for things like traitor, so on?

The justification for them not being able to roll antag was round start sec access, but that's fixed now, yeah?

Majkl-J commented:
This whole PR will need considerable amount of discussion. I'm usually against removing stuff over admin issues, twice as much when I haven't really seen the playerbase abuse it aside from rare isolated incidents, so I'm not entirely convinced here.

> If they're getting security access revoked, can they be put back in the antag list for things like traitor, so on?

Good question, will think about it.

nikothedude commented:
admin issue? this is a design issue, guards were never meant to have access to sec nor sec records, the pr doesnt even mention admin issues

Majkl-J commented:
> admin issue? this is a design issue, guards were never meant to have access to sec nor sec records, the pr doesnt even mention admin issues

Whatever the ancient intended design purpose for them was 3 years ago on a server where nothing ever happened, it does not  apply to Bubberstation.

What is an actual issue: Dep guards aren't properly outlined in the rules as far as I know (There is a total of one mention of them in spacelaw, which outlines them as falling under Sxx together with security), which is something that probably needs to be addressed before any actual code change is made.

Sure they aren't full on security. But they also deal with problems that security is meant to deal with, within their department.

CatoChristopherMrow commented:
> Sure they aren't full on security. But they also deal with problems that security is meant to deal with, within their department.
Sure and that's perfectly fine, but you don't need to be able to set people to wanted or go into security. Guards purpose is to defend their department and kick people out if need be, neither of that requires sec access.

> What is an actual issue: Dep guards aren't properly outlined in the rules as far as I know (There is a total of one mention of them in spacelaw, which outlines them as falling under Sxx together with security), which is something that probably needs to be addressed before any actual code change is made.

The only thing that's for sure is that they're NOT security, so I don't feel too bad removing this access.

This PR really won't effect them too much other then them no longer being able to go into sec, sec no longer being able to go into their offices, and no longer being able to set people to wanted and edit sec records.

Iamgoofball commented:
>Whatever the ancient intended design purpose for them was 3 years ago on a server where nothing ever happened, it does not apply to Bubberstation.

I played both Security and Guards at bubber and very much can confirm that the exact issues that this PR resolves were absolutely occuring as of 1 week ago. This is still an issue and this PR still needs to happen.

Kajortoq commented:
I’m HoS Lightfeather and this is my favorite PR on the github.

YakumoChen commented:
Security should still get access to guard outposts, but not the lockers imo

As a regular HoS I often use outposts when making arrests in departments for holding/pre-move to security, it is more convenient and otherwise would remove a lot of access to gun chargers, for example.

It's not the biggest deal if its removed but I don't see access to the room/charger as a point of contention very often compared to access to the department as a whole

CatoChristopherMrow commented:
> Security should still get access to guard outposts, but not the lockers imo

Yeah, I was originally going to do it like this but the way mapping helpers is setup it requires all depguard access to enter it. HoSes should still theoretically be able to access the outposts, just not normal secoffs. Given that we're removing the guards sec access, I find it decently fair we don't get their stuff either, to further reinforce they're not sec. This was also goofs original vison, so I really don't mind too much.

DorfGuy1 commented:
> > Whatever the ancient intended design purpose for them was 3 years ago on a server where nothing ever happened, it does not apply to Bubberstation.
> 
> I played both Security and Guards at bubber and very much can confirm that the exact issues that this PR resolves were absolutely occuring as of 1-2 weeks ago. This is still an issue and this PR still needs to happen. Players still mistakenly believe they're Security because their ID cards let them into Security and their office looks like Security. This PR resolves that.
> 
> Guards are not Security. They do not answer to the HoS. They do not make arrests. They are paul blart mall cops in the most literal sense. Their job is to kick people out of departments and reclaim stolen goods in the process. There's a reason their batons aren't infinite use outside of their departmental areas.
> 
> I, very regularly, had to argue with Guard players about them not being Security when they used their access to roll into our equipment rooms, print a bunch of shit they aren't supposed to have, and take MCRs/laser carbines/disabler SMGs, along with the aformentioned "taking wanted status off of their friends" problem. Skyrat fucked up my design when they refused to re-add the offices and this caused these problems to manifest.

Which offices did they fail to re-add? The security ones for actual security in various departments?

CatoChristopherMrow commented:
> Which offices did they fail to re-add? The security ones for actual security in various departments?

https://github.com/Skyrat-SS13/Skyrat-tg/pull/17171

Skyrat refused to do automapper stuff, for some reason, so they were forced to give guards sec access.

NanoCats commented:
> This was also goofs original vison, so I really don't mind too much.

I don't think Goof is a golden god who should have unilateral control to exert his "vision" over our codebase, just putting it out there. 

Echoing Waterpig a bit, personally I'd rather guards be _more_ integrated as opposed to this "no girls allowed" attitude people whip out when they're like "nyeerrrr you're not sec!!!!!".  They are still sec adjacent at the very least, and no, **I don't care how many times goof or anyone else for that matter recites the phrase "they aren't sec"**. Like there isn't even a way to priority-radio in bad things bad people do in my department, and on common radio things get drowned out easily by endless braying from the general crew.  Sec players need to stop being elitist and remember that the department guards also want to have fun and not just stand in a corner. 
 
I believe there are better ways this "problem" could be dealt with, like the sec consoles checking trim if you're that fussed about records. 

As a point of order; guards can't go into the locker room on all current maps because they have different access. If they're doing that, then you can just, eject them. 

If you want to do something cool how about fixing paramedic access?  Or is that not going to happen because this PR was made for revenge or to punish, not just to remedy a real problem that needs solved.  Like if someone is stealing your guns

1. Don't leave guns all over the floor, I don't know why Bubber has such a big problem with this, but people are way too loose with leaving guns laying around in sec, they should be in the locker room or the armory. People will just, open a gun crate from cargo and leave it open on the floor 💀. I've taken the time to make sure guns were at least close to locked up properly and people acted like I had disintegrated them when they had to put 5% thought into locating them; guns, in the armory? What an idea.

2. Prosecute them for stealing like you would any other crime, do your security job security boy. 

All this is going to change is make it more annoying for me to get magazines for my gun because a divine edict from the heavens has decreed the only way to get more magazines (without buying a whole separate firearm) is sec's lathe. Oh and I guess I can't buy sunglasses anymore.  Absolutely _cannot wait_ for the conversation where a warden is like "Erm you don't need this" when I ask to print out a magazine for a gun that I'm allowed to have 🙄

For the record I firmly believe this is a salt PR because I set a valid fine on a paramedic for trespassing, (sheep anthro abusing their access as a paramedic to walk up to me, bleat, and walk back out before I can do anything to them).  If a crime like this happens it's pretty common for sec to do nothing about it. Paras can abuse their access relentlessly and this has been going on for... How long have paramedics been a thing on Tgcode? About that long.  Needless to say I'm considering this a personal attack.

Basically "skill issue"/IC issue as the annoying people I hate say.

Iamgoofball commented:
>For the record I firmly believe this is a salt PR because I set a valid fine on a paramedic for trespassing, (sheep anthro abusing their access as a paramedic to walk up to me, bleat, and walk back out before I can do anything to them)

You're not supposed to issue them fines as a Guard, you're supposed to just boot them the fuck out by force using your weapons.

NanoCats commented:
> > For the record I firmly believe this is a salt PR because I set a valid fine on a paramedic for trespassing, (sheep anthro abusing their access as a paramedic to walk up to me, bleat, and walk back out before I can do anything to them)
> 
> You're not supposed to issue them fines as a Guard, you're supposed to just boot them the fuck out by force using your weapons. I feel like you don't fundamentally understand the underlying design here.

Yeah but paramedics will just dip in, do crime, and fuck off before you have time to so much as tell them to fuck off.

CatoChristopherMrow commented:
> Echoing Waterpig a bit, personally I'd rather guards be _more_ integrated as opposed to this "no girls allowed" attitude people whip out when they're like "nyeerrrr you're not sec!!!!!". They are still sec adjacent at the very least, and no, **I don't care how many times goof or anyone else for that matter recites the phrase "they aren't sec"**. Like there isn't even a way to priority-radio in bad things bad people do in my department, and on common radio things get drowned out easily by endless braying from the general crew. Sec players need to stop being elitist and remember that the department guards also want to have fun and not just stand in a corner.

Yes there is, there is a security emergency button on all requests consoles.

> As a point of order; guards can't go into the locker room on all current maps because they have different access. If they're doing that, then you can just, eject them.

They can on plenty of maps, because of their access and how tg mapped some maps. This fixes this.

> If you want to do something cool how about fixing paramedic access? Or is that not going to happen because this PR was made for revenge or to punish, not just to remedy a real problem that needs solved. Like if someone is stealing your guns

I plan on it next. Please stop being so aggressive, github is not the place for this kinda comment.

> 2. Prosecute them for stealing like you would any other crime, do your security job security boy.

Again, github is not the place for this kind of comment. You're ignoring all the other issues other then just theft.

> For the record I firmly believe this is a salt PR because I set a valid fine on a paramedic for trespassing, (sheep anthro abusing their access as a paramedic to walk up to me, bleat, and walk back out before I can do anything to them). If a crime like this happens it's pretty common for sec to do nothing about it. Paras can abuse their access relentlessly and this has been going on for... How long have paramedics been a thing on Tgcode? About that long. Needless to say I'm considering this a personal attack.

I don't even know who you are. This is a change I've been vocal about for quite some time.

> Basically "skill issue"/IC issue as the annoying people I hate say.
Again, github is not the kind of place for comments like this.

ReturnToZender commented:
> Yeah but paramedics will just dip in, do crime, and fuck off before you have time to so much as tell them to fuck off.
Out of scope, this PR is to separate security access from departmental guards

Use this to send a priority alert to Security when bad things happen in your department, each department has one and you can use it to call more than Security
![image](https://github.com/user-attachments/assets/1a108537-54f0-48fb-a30c-68c48d4efa7d)

(Legally required to add self-report in response to these comments)

Kajortoq commented:
Weighing on previous discussion, integrating guards more might not hurt and honestly pretty alright. 

Integrating them namely by making them subordinates to the HoS akin to corrections officers for departments as opposed to answering to that department’s head. 

But that’s asking a lot and separating them from sec seems more viable such as in this PR.

Iamgoofball commented:
>Yeah but paramedics will just dip in, do crime, and fuck off before you have time to so much as tell them to fuck off.

Normalize beating the shit out of Para-tiders when they do this. Just straight up kill them on the spot when they trespass. Get your fellow department buddies in on the beating.

CatoChristopherMrow commented:
> Integrating them namely by making them subordinates to the HoS akin to corrections officers for departments as opposed to answering to that department’s head.

That's just depsec again

Kajortoq commented:
> > Integrating them namely by making them subordinates to the HoS akin to corrections officers for departments as opposed to answering to that department’s head.
> 
> That's just depsec again

I mean yeah basically. It’d be department sec but with less access and powers. That’s not the focus of this PR though and I support this PR’s intention as is.

YakumoChen commented:
> Sec players need to stop being elitist and remember that the department guards are actual people who also want to have fun and not just stand in a corner.

This isn't a problem with sec elitism, the actual problem is giving guards more tools to be like security or communicate is just a unilateral buff to security and validhunting

CatoChristopherMrow commented:
(mostly with more clear guard rules, since it's not clearly defined and most people don't know them)

NargaInDisguise commented:
> If they're getting security access revoked, can they be put back in the antag list for things like traitor, so on?
> 
> The justification for them not being able to roll antag was round start sec access, but that's fixed now, yeah?
> 
> (Speaking as someone who wants to play heretic or traitor as a dept guard sometimes lol)

Personally I'm very against re-adding them as antags, we saw it during the couple weeks where it was broken and giving antags to all blacklisted roles and they got away with literally anything they wanted because they had a modicum of authority over the department that was abused to the extreme. Just remove their sec access and make them first priority on deputizations, don't add them back as antags, please.

Formerxboxgamer commented:
> > Integrating them namely by making them subordinates to the HoS akin to corrections officers for departments as opposed to answering to that department’s head.
> 
> That's just depsec again

What if we just removed guards because they were a code solution to an administrative problem of security having a monopoly on antag rp on Skyrat.

NargaInDisguise commented:
> > > Integrating them namely by making them subordinates to the HoS akin to corrections officers for departments as opposed to answering to that department’s head.
> > 
> > 
> > That's just depsec again
> 
> What if we just removed guards because they were a code solution to an administrative problem of security having a monopoly on antag rp on Skyrat.

I like orderly :(

sippykot commented:
so, how is the service guard supposed to recharge their stuff now? they dont have a guard station on most maps

sippykot commented:
> > > Integrating them namely by making them subordinates to the HoS akin to corrections officers for departments as opposed to answering to that department’s head.
> > 
> > 
> > That's just depsec again
> 
> What if we just removed guards because they were a code solution to an administrative problem of security having a monopoly on antag rp on Skyrat.

its really useful to have someone in the department to actually defend it at least sometimes
with department guards you have backup sometimes, compared to the default with the security stations never being manned and seccies being complete outsiders
also gives heads some muscle aside from having to call sec over or beat up people personally

Swiftfeather commented:
> > > > Integrating them namely by making them subordinates to the HoS akin to corrections officers for departments as opposed to answering to that department’s head.
> > > 
> > > 
> > > That's just depsec again
> > 
> > 
> > What if we just removed guards because they were a code solution to an administrative problem of security having a monopoly on antag rp on Skyrat.
> 
> I like orderly :(

I personally think all of them except orderly should be removed. Orderly makes sense because the medbay do be kinda bumpin.

I eng dep guards usually just do obnoxious projects or go erp instead of guarding. Idek why we have a bouncer when the bartender has a traditional shotgun. Cargo will have the odd miner and gun orders. Everybody ignores Sci it's not real unless you're a ghostspawn antag

Sebastionate commented:
Helo I have like 400 hours in Engi guard and it's my main role nowadays

Removing guard would be bum as fuck and the reason stated for removing it are personal bias/purely personally experienced. 

I personally, _outside of 'being useless and erping'_, do often prevent a lotta antaggery/trespass in the department itself, and having someone always hovering around the dept-desk to handle personal orders (for any department really) is a great thing

I have qualms with removing access to sec records personally, as said it's abused rarely and abuse of such is violation of OOC rules, just the same as Sec being blatantly corrupt/ruining rounds for others via their additional power

StrangeWeirdKitten commented:
I’ll make the decision later

aKromatopzia commented:
this PR is conceptually silly. Why should a guard be unable to update records? Make them unable to negate or update records sure, but to be barred from setting any at all is horrid. I know that for the longest time AI functioned like that.

As it already stands, nobody enforces basic IC law except the guards, because sec won't be deploying for basic trespass or basic bne. Taking this power away from the people who explicitly want to be Paul Blarts doing free labor for sec is just weird.

Kajortoq commented:
> this PR is conceptually silly. Why should a guard be unable to update records? Make them unable to negate or update records sure, but to be barred from setting any at all is horrid. I know that for the longest time AI functioned like that.
> 
> As it already stands, nobody enforces basic IC law except the guards, because sec won't be deploying for basic trespass or basic bne. Taking this power away from the people who explicitly want to be Paul Blarts doing free labor for sec is just weird.

Half of what sec does is arrest for trespass and breakins and run over for that. If sec doesn’t respond, it’s probably because it wasn’t loudly stated in common and sec missed it, the department head didn’t ask the HoS for help and sec is unaware it’s needed, or sec is being slightly stomped on by an antag. 

With this, guards would still be able to defend and handle their departments. They just won’t be in records and loitering in sec asking for sec equipment. Guards don’t typically do full arrests as is.

StrangeWeirdKitten commented:
I asked the other maintainers and the ultimate decision was a pretty big resounding no. Ahelp if a depguard is overextending their role.

CatoChristopherMrow commented:
> I asked the other maintainers and the ultimate decision was a pretty big resounding no. Ahelp if a depguard is overextending their role.

Could I ask exactly why? I'd love to know the reasoning so I could potentially make some changes.

Iamgoofball commented:
This is a really bad decision. Do better, or just remove Guards if you're going to let my work exist in such an unmaintained state.

StrangeWeirdKitten commented:
> > I asked the other maintainers and the ultimate decision was a pretty big resounding no. Ahelp if a depguard is overextending their role.
> 
> Could I ask exactly why? I'd love to know the reasoning so I could potentially make some changes.

Primarily because it was opted to just be handled administratively for the time being because it's not really an issue or trap people fall into. If someone is a poor department guard, they tend to have a lot of note history following them. This is what I told Goof. TD;LR, let's let department guards handle smaller crimes and cite people. They should have some basic security access so if someone is really stressing their department, they can walk the dude up in cuffs to the brig (which I also find extremely valid)

![image](https://github.com/user-attachments/assets/a834f74a-f8f7-4941-8a7d-b32004ec3b57)

aKromatopzia commented:
> Half of what sec does is arrest for trespass and breakins and run over for that. If sec doesn’t respond, it’s probably because it wasn’t loudly stated in common and sec missed it, the department head didn’t ask the HoS for help and sec is unaware it’s needed, or sec is being slightly stomped on by an antag.

I can tell you that this doesn't happen when I watch the dept, but I can chalk this up to playerbase and timezone differences

CatoChristopherMrow commented:
> > > I asked the other maintainers and the ultimate decision was a pretty big resounding no. Ahelp if a depguard is overextending their role.
> > 
> > 
> > Could I ask exactly why? I'd love to know the reasoning so I could potentially make some changes.
> 
> Primarily because it was opted to just be handled administratively for the time being because it's not really an issue or trap people fall into. If someone is a poor department guard, they tend to have a lot of note history following them. This is what I told Goof. TD;LR, let's let department guards handle smaller crimes and cite people. They should have some basic security access so if someone is really stressing their department, they can walk the dude up in cuffs to the brig (which I also find extremely valid)
> 
> Also their is niche policy on the wiki. (granted it's not in the most obvious place) But this can honestly be added to their in game job blurb.

Perhaps something closer to explicitly letting them fine people (and nothing else) and giving them ONLY front door sec access would be more to your tastes as a PR then?

NargaInDisguise commented:
> 
> Perhaps something closer to explicitly letting them fine people (and nothing else) and giving them ONLY front door sec access would be more to your tastes as a PR then?

Cato you and I both know this makes even less sense. If an orderly watches someone stab someone to death in the lobby and gets involved because sec is currently preoccupied, they're not just gonna go "sorry bubs I can only fine them 100 credits for vandalizing our floors with the blood of someone," that is EXTREMELY silly. That's not how security records should or need to work. Just actually let admins do their jobs and weed out bad actors.

Guards can make arrests if they're within the immediate vicinity of the crime, but they don't have the rights to go chasing after someone if they escape. That's what a guard is, they protect their area.

aKromatopzia commented:
(secondary note, if security is arresting purely off sec records without checking who or why the note was placed (by), maybe the fault is on them too for not communicating anyways. this case in particular only applies of course to situations where a person isn't caught red handed doing antaggery)
